app/src/lib/stores/git-store.ts

```diff
@@ -891,15 +891,18 @@ export class GitStore extends BaseStore {
 
     if (currentBranch || currentTip) {
       if (currentTip && currentBranch) {
-        const cachedCommit = this.commitLookup.get(currentTip)
-        const branchTipCommit =
-          cachedCommit ||
-          (await this.performFailableOperation(() =>
-            getCommit(this.repository, currentTip)
-          ))
-
-        if (!branchTipCommit) {
-          throw new Error(`Could not load commit ${currentTip}`)
+        let branchTipCommit = this.commitLookup.get(currentTip) || null
+        if (branchTipCommit == null) {
+          branchTipCommit =
+            (await this.performFailableOperation(() =>
+              getCommit(this.repository, currentTip)
+            )) || null
+
+          if (branchTipCommit != null) {
+            this.commitLookup.set(currentTip, branchTipCommit)
+          } else {
+            throw new Error(`Could not load commit ${currentTip}`)
+          }
         }
```

Comment:
Could this section become just:

```ts
let branchTipCommit =
  this.commitLookup.get(currentTip) ||
  (await this.performFailableOperation(() =>
    getCommit(this.repository, currentTip)
  )) ||
  null;

if (branchTipCommit == null) {
  throw new Error(`Could not load commit ${currentTip}`);
} else {
  this.commitLookup.set(currentTip, branchTipCommit);
}
```

Reply:
Maybe? I'm just over here quietly crying about having to deal with `null` and `undefined` here but I might have to suck it up and always call `this.commitLookup.set` (should be safe).